src/Hosting/Environment.php

```diff
@@ -100,7 +100,7 @@ public function getName()
      */
     public function getRevision()
     {
-        if ($this->revision === null) {
+        if (empty($this->revision)) {
```

Comment:
@scottreese actually as I look at the behavior in the related support-platform-cloud-data-model PR, I wonder if this should remain a check for null? If `$this->revision` has never been set then it would remain initialized as null; if it was set to an empty string then it would return as such, and you wouldn't need to catch the exception in `InternalEnvironment::getArrayCopy()`. Thoughts?

Reply:
Yeah actually thinking back to the original design, I think the intention of the getter exception is to indicate that a property was never set; and the setter exception to indicate attempts to set bad values.

Reply:
Ah, ok, that makes sense. I was thinking about it from the perspective of covering both instances where the statement "This Environment object does not know its VCS revision" would be true. Maybe I was being a bit too literal there. I did go back and forth about this in my head so I think I'll go ahead and put it back to a null check.

Reply:
I reverted this change and confirmed that the change to support-platform-cloud-data-model is no longer needed. Thanks!